server/src/main/java/org/opensearch/search/approximate/ApproximatePointRangeQuery.java

```diff
@@ -293,63 +292,45 @@ public void intersectLeft(PointValues.IntersectVisitor visitor, PointValues.Poin
                 }
             }
 
-            // custom intersect visitor to walk the right of tree
+            // custom intersect visitor to walk the right of tree (from rightmost leaf going left)
             public void intersectRight(PointValues.IntersectVisitor visitor, PointValues.PointTree pointTree, long[] docCount)
                 throws IOException {
                 PointValues.Relation r = visitor.compare(pointTree.getMinPackedValue(), pointTree.getMaxPackedValue());
                 if (docCount[0] >= size) {
                     return;
                 }
                 switch (r) {
-                    case CELL_OUTSIDE_QUERY:
-                        // This cell is fully outside the query shape: stop recursing
-                        break;
-
                     case CELL_INSIDE_QUERY:
-                        // If the cell is fully inside, we keep moving right as long as the point tree size is over our size requirement
-                        if (pointTree.size() > size && docCount[0] < size && moveRight(pointTree)) {
+                    case CELL_CROSSES_QUERY:
+                        if (pointTree.moveToChild() && docCount[0] < size) {
+                            while (pointTree.moveToSibling()) {
```

Comment:
No need use the `while`, there be only on sibling node

Reply:
Replace `pointTree.moveToSibling()`with `while (pointTree.moveToSibling()) {}`.

Reply:
@kkewwei I understood so you are suggesting to remove `while (pointTree.moveToSibling()) {}` and just keep `pointTree.moveToSibling();` ?

I followed the `intersectLeft` implementation which uses a loop. Can we move forward with the same implementation while I read more on the point `there be only on sibling node` ?

Reply:
Thanks @msfroh make sense to me I have updated the code to remove the while loop and added a comment mentioning about the scenario.